Sources/System/FilePath/FilePathString.swift

```diff
@@ -20,6 +20,30 @@ extension FilePath {
     self.init(_platformString: platformString)
   }
 
+  public init(platformString: [CInterop.PlatformChar]) {
+    guard let _ = platformString.firstIndex(of: 0) else {
+      preconditionFailure(
+        "input of FilePath.init(platformString:) must be null-terminated"
+      )
+    }
+    self = platformString.withUnsafeBufferPointer {
+      FilePath(_platformString: $0.baseAddress!)
+    }
+  }
+
+  public init(platformString: inout CInterop.PlatformChar) {
```

Comment:
What role is this init meant to serve?

Reply:
In the stdlib, I marked the equivalent as deprecated with a suggestion to use `String.init(_: UnicodeScalar)` instead.

Reply:
Can we do deprecation too?

Reply:
I think we can. I did want your opinion on that.